Source/SwiftLintFramework/Reporters/GroupedReporter.swift

```diff
@@ -0,0 +1,88 @@
+import SwiftyTextTable
+
+public struct GroupedReporter: Reporter {
+    // MARK: - Reporter Conformance
+
+    public static let identifier = "grouped"
+    public static let isRealtime = false
+
+    public var description: String {
+        return "Reports how many times certain violations are broken."
+    }
+
+    private static func groupViolations(_ violations: [StyleViolation],
+                                        WithViolationsSeverity severity: ViolationSeverity) -> [[StyleViolation]] {
```

Comment:
It got me a while to grasp this method. It can be entirely replaced by
```swift
let groupedBySeverity = violations
    .group(by: \.severity)
    .mapValues { $0.group(by: \.ruleIdentifier).sorted(by: { $0.value.count > $1.value.count }) }

let groupedErrors = groupedBySeverity[.error] ?? []
let groupedWarnings = groupedBySeverity[.warning] ?? []
```
where `groupedErrors` and `groupedWarnings` can be counted and passed on to create the table.

Reply:
Hi,

I either did not follow what you did here or it was not quite what I wanted. I did change the approach to make it more readable though. I hope you like it or you have a better suggestion.